### PR TITLE
Pop open the share modal when the user navigates to a recording with /share

### LINKED ISSUE #5163: Add routing so that we have a way to automatically open the sharing modal upon opening a replay
Follow up to #5153.

This allows us to send the replay owner a notification when somebody requests access to their replay with a link that opens their replay, and pops open the sharing modal immediately after.

### PR DESCRIPTION
**Pull request by jaril**
Fix #5163.

### CODE REVIEW

**Review by ryanjduffy — COMMENTED** [review]
I worry about overloading the `[id]` route with all possible future subroutes. Did you try adding ` [id]/share` route? That would be easier to extend if it's possible.

File: pages/recording/[id]/share.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useRouter } from "next/router";
+import { useDispatch } from "react-redux";
+import { setModal } from "ui/actions/app";
+import { useGetRecordingId } from "ui/hooks/recordings";
+
+export default function Share() {
+  const recordingId = useGetRecordingId();
```
Comment (ryanjduffy) [inline pages/recording/[id]/share.tsx:7]: Can probably drop this since you're pulling the recording id from the query too.

**Review by ryanjduffy — APPROVED** [review]

**Comment by ryanjduffy** [comment]
I'll add that I think using `replace()` is appropriate until we start including modals in our routes otherwise. Since clicking the Share button doesn't update the route yet, it makes sense to me to drop `/share` for this case.

File: pages/recording/[id]/share.tsx (hunk 1 above)
Comment (jaril) [reply pages/recording/[id]/share.tsx]: left it on just in case we have to handle slugs

**Comment by gideonred** [comment]
FYI, if the replay-soc2 bot is annoying here what we could do is add logic to it to not approve draft PRs and you guys just need to flag the PR as draft.

**Comment by gideonred** [comment]
or, I guess I could just have it check it has already been approved.

File: pages/recording/[id]/share.tsx (hunk 1 above)
Comment (ryanjduffy) [reply pages/recording/[id]/share.tsx:7]: gotcha :+1:

**Review by ryanjduffy — APPROVED** [review]

**Pull request merged**